hw/ip/prim/README.md

```diff
@@ -32,198 +32,165 @@
+  "lowrisc:prim:clock_inv"    : lowrisc:prim_generic:clock_inv
+  "lowrisc:prim:clock_mux2"   : lowrisc:prim_xilinx:clock_mux2
+  # ...
 ```
 
+Notice that there aren't xilinx specific implementations for all primitives, so the generic implementation is requested.
 
-### Set the default technology library
+Beyond primitives, mappings are also used in the top-level core files, e.g. in `hw/top_earlgrey/top_earlgrey.core`, to ask for top specific implementations.
 
-If no specific technology library is chosen for an instantiated primitive the
-default library is used. The SystemVerilog define `PRIM_DEFAULT_IMPL` can be
-used to set the default for the whole design. Set this define to one of the enum
-values in `prim_pkg.sv` in the form `prim_pkg::ImplTechlibname`. `Techlibname`
-is the capitalized name of the technology library.
+A specific use of mappings is the lints for each block, for example `hw/ip/uart/uart.core`'s lint target.
+None of it's dependencies contain concrete primitive implementations.
+Instead mappings are used when dvsim invokes the core.
+This allows the block to be linted for multiple different primitives (and top-level constants).
 
-In the top-level FuseSoC core file the default technology library can be chosen
-like this:
+[mappings]: https://fusesoc.readthedocs.io/en/stable/user/build_system/mappings.html
 
-```yaml
-# my_toplevel.core
-
-# Declare filesets and other things (omitted)
-
-parameters:
-  # Make the parameter known to FuseSoC to enable overrides from the
-  # command line. If not overwritten, use the generic technology library.
-  PRIM_DEFAULT_IMPL:
-    datatype: str
-    paramtype: vlogdefine
-    description: Primitives implementation to use, e.g. "prim_pkg::ImplGeneric".
-    default: prim_pkg::ImplGeneric
-
-targets:
-  fpga_synthesis:
-    filesets:
-      - my_rtl_files
-    parameters:
-      # Use the xilinx technology library for this target by default.
-      - PRIM_DEFAULT_IMPL=prim_pkg::ImplXilinx
-    toplevel: my_toplevel
-```
+
+## User Guide
+### Using primitives
+
+Primitives are normal SystemVerilog modules, and can be used as usual:
+1. Instantiate it like a normal SystemVerilog module.
+   ```systemverilog
+   prim_fifo_sync #(
+     .Width   (8),
+     .Pass    (1'b0),
+     .Depth   (TxFifoDepth)
+   ) u_uart_txfifo (
+     .clk_i,
+     // ..
+     .err_o   ()
+   )
+   ```
+2. Add it as a dependency in the FuseSoC core file.
+   ```yaml
+   name: "lowrisc:ip:uart:0.1"
+   description: "uart"
+   filesets:
+     files_rtl:
+       depend:
+         - lowrisc:virtual_constants:top_pkg
+         - lowrisc:prim:prim_fifo_sync
+   ```
+
+
+If you'd like to depend on a specific implementation, you can add the specific implementation's VLNV instead of the virtual VLNV.
+For example, you would depend on `lowrisc:prim_xilinx:prim_fifo_sync` instead of `lowrisc:prim:prim_fifo_sync` if you wanted to forcefully use the Xilinx implementation.
+**This is not recommended.**
```

Comment:
nit: Best to provide a reason alongside recommendations.

```suggestion
Because this method can restrict the flexibility to swap compatible cores, **this is not recommended.**
```